From 320bea0b9358fae2fc216b82137757bb4f52cc61 Mon Sep 17 00:00:00 2001
From: Jean-Philippe ROMAIN <jean-philippe.romain@st.com>
Date: Mon, 8 Jul 2019 16:10:45 +0200
Subject: [PATCH 1/2] Move dtimg_addr down of 128 bytes (header size) in order
 to align dt_addr on page size (4KB)


diff --git a/include/configs/stm32mp1.h b/include/configs/stm32mp1.h
index cb40b77..76bd875 100644
--- a/include/configs/stm32mp1.h
+++ b/include/configs/stm32mp1.h
@@ -166,7 +166,7 @@
 
 #define STM32MP_ANDROID \
 	"suffix=a\0" \
-	"dtimg_addr=0xc4500000\0" \
+	"dtimg_addr=0xc44FFF80\0" \
 	"android_mmc_splash="\
 		"if part start mmc ${devnum} splash splash_start && " \
 		   "part size mmc ${devnum} splash splash_size;"\
-- 
2.7.4

