From 37d3c66410857caedfa9f74ebfa4f60ae394c1ad Mon Sep 17 00:00:00 2001
From: frq09432 <nicolas.louboutin@st.com>
Date: Fri, 11 Oct 2019 13:43:13 +0200
Subject: [PATCH 2/2] stm32mp1: boot with ramdisk (integrated in boot image)

---
 include/configs/stm32mp1.h | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/include/configs/stm32mp1.h b/include/configs/stm32mp1.h
index fd0566a..20b5c95 100644
--- a/include/configs/stm32mp1.h
+++ b/include/configs/stm32mp1.h
@@ -186,7 +186,6 @@
 		   "mmc read ${kernel_addr_r} ${boot_start} ${boot_size};" \
 		   "part nb mmc ${devnum} system_${suffix} rootpart_nb;" \
 		   "env set bootargs " \
-		     "root=/dev/mmcblk${devnum}p${rootpart_nb} " \
 		     "androidboot.serialno=${serial#} " \
 		     "androidboot.optee=true " \
 		     "androidboot.slot_suffix=_${suffix};"\
@@ -200,7 +199,6 @@
 		   "mmc read ${kernel_addr_r} ${boot_start} ${boot_size};" \
 		   "part nb mmc ${devnum} system_${suffix} rootpart_nb;" \
 		   "env set bootargs " \
-		     "root=/dev/mmcblk${devnum}p${rootpart_nb} " \
 		     "androidboot.serialno=${serial#} " \
 		     "androidboot.slot_suffix=_${suffix};"\
 		"fi\0"
@@ -230,7 +228,7 @@
 		"run android_mmc_splash;" \
 		"run android_mmc_fdt;" \
 		"run android_mmc_kernel;" \
-		"bootm ${kernel_addr_r} - ${fdt_addr_r};\0" \
+		"bootm ${kernel_addr_r} ${kernel_addr_r} ${fdt_addr_r};\0" \
 	"bootcmd_android=" \
 		"env set mmc_boot run android_mmc_boot;" \
 		"run bootcmd_stm32mp\0"
-- 
2.7.4

